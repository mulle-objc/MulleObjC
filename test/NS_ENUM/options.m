#ifndef __MULLE_OBJC__
# import <Foundation/Foundation.h>
#else
# import <MulleObjC/MulleObjC.h>
#endif


typedef NS_OPTIONS( short, Foo)
{
   FooA = 0x01,
   FooB = 0x04,
   FooC = 0x10
};


extern NS_OPTIONS_TABLE( Foo, 3);


NS_OPTIONS_TABLE( Foo, 3) =
{
   NS_OPTIONS_ITEM( FooA),
   NS_OPTIONS_ITEM( FooB),
   NS_OPTIONS_ITEM( FooC)
};


int   main( void)
{
   printf( "%s\n", NS_OPTIONS_PRINT( Foo, FooA));
   printf( "%s\n", NS_OPTIONS_PRINT( Foo, FooB|FooC));
   printf( "%s\n", NS_OPTIONS_PRINT( Foo, FooA|FooB|FooC));

   printf( "%s\n", NS_OPTIONS_PRINT( Foo, NS_OPTIONS_PARSE( Foo, "FooA")));
   printf( "%s\n", NS_OPTIONS_PRINT( Foo, NS_OPTIONS_PARSE( Foo, "FooB|FooC")));
   printf( "%s\n", NS_OPTIONS_PRINT( Foo, NS_OPTIONS_PARSE( Foo, "FooA|FooB|FooC")));

   return( 0);
}
