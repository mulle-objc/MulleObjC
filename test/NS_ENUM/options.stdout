FooA
FooB|FooC
FooA|FooB|FooC
FooA
FooB|FooC
FooA|FooB|FooC
